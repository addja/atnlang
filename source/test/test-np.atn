def isUpper (s) {
	return s[0] >= "A" and s[0] <= "Z";
}

def member (s, l) {
	for (i = 0; i < l.length; i = i +1) {
		if (s == l[i]) return true;
	}
	return false;
}

pre[1] = "dr";
pre[0] = "sr";
fun[2] = "de";
fun[1] = "la";
fun[0] = "el";
suf[1] = "junior";
suf[0] = "senior";

atn np {
	node IN {
		arc (isUpper(#.0)) goto NP;
		arc (member(#.0, pre)) goto PREF {
			print pre, "%n";
			print #.0;
		}
	}
	node PREF {
		arc (member(#.0, pre)) goto PREF;
		#peta a isUpper !!
		arc (isUpper(#.0)) goto NP;
	}
	node NP {
		arc (isUpper(#.0)) goto NP;
		arc (member(#.0, suf)) goto SUF;
		arc (member(#.0, fun)) goto FUN;
		arc (true) goto END;
	}
	node SUF {
		arc (member(#.0, suf)) goto SUF;
		arc (true) goto END;
	}
	node FUN {
		arc (member(#.0, fun)) goto FUN;
		arc (true) goto END;
	}
	node END accept;
}	

def main() {
	a = atn(np);		
	print a;
}
