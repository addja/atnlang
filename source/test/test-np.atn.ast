(PROGRAM (def isUpper (PARAMS s) (LIST_INSTR (return (and (>= ([ s 0) "A") (<= ([ s 0) "Z"))))) (def member (PARAMS s l) (LIST_INSTR (for (:= i 0) (< i (ARRAYLENGTH l)) (:= i (+ i 1)) (LIST_INSTR (if (== s ([ l i)) (LIST_INSTR (return true))))) (return false))) (:= nombre "") (atn np (NODELIST (:= pre (ARRAY_DECL "dr" "sr" "de")) (:= fun (ARRAY_DECL "de" "la" "el")) (:= suf (ARRAY_DECL "junior" "senior")) (node IN (ARC_LIST (arc (FUNCALL isUpper (ARGLIST (# 0))) (goto NP) (LIST_INSTR (:= nombre (+ (+ nombre (# 0)) " ")))) (arc (FUNCALL member (ARGLIST (# 0) pre)) (goto PREF) (LIST_INSTR (:= nombre (+ (+ nombre (# 0)) " ")))) (arc true (goto IN)))) (node PREF (ARC_LIST (arc (FUNCALL member (ARGLIST (# 0) pre)) (goto PREF) (LIST_INSTR (:= nombre (+ (+ nombre (# 0)) " ")))) (arc (FUNCALL isUpper (ARGLIST (# 0))) (goto NP) (LIST_INSTR (:= nombre (+ (+ nombre (# 0)) " ")))) (arc true (goto IN) (LIST_INSTR (:= nombre ""))))) (node NP (ARC_LIST (arc (FUNCALL isUpper (ARGLIST (# 0))) (goto NP) (LIST_INSTR (:= nombre (+ (+ nombre (# 0)) " ")))) (arc (FUNCALL member (ARGLIST (# 0) suf)) (goto SUF) (LIST_INSTR (:= nombre (+ (+ nombre (# 0)) " ")))) (arc (FUNCALL member (ARGLIST (# 0) fun)) (goto FUN) (LIST_INSTR (:= nombre (+ (+ nombre (# 0)) " ")))) (arc true (goto END)))) (node SUF (ARC_LIST (arc (FUNCALL member (ARGLIST (# 0) suf)) (goto SUF) (LIST_INSTR (:= nombre (+ (+ nombre (# 0)) " ")))) (arc true (goto END)))) (node FUN (ARC_LIST (arc (FUNCALL member (ARGLIST (# 0) fun)) (goto FUN) (LIST_INSTR (:= nombre (+ (+ nombre (# 0)) " ")))) (arc (FUNCALL isUpper (ARGLIST (# 0))) (goto NP) (LIST_INSTR (:= nombre (+ (+ nombre (# 0)) " ")))))) (node END accept))) (def main PARAMS (LIST_INSTR (:= a (ATNCALL np)) (if a (LIST_INSTR (print "Nombre encontrado: " nombre "%n")) (LIST_INSTR (print "ningÃºn nombre ha sido encontrado" "%n"))))))