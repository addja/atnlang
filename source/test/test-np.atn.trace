main() <entry point>
|   np() <line 69>
|   |   isUpper(s=de) <line 20>
|   |   return false <line 2>
|   |   member(s=de, l=[dr,sr,de]) <line 23>
|   |   return true <line 6>
|   |   member(s=entre, l=[dr,sr,de]) <line 29>
|   |   return false <line 9>
|   |   isUpper(s=entre) <line 32>
|   |   return false <line 2>
|   |   isUpper(s=todos) <line 20>
|   |   return false <line 2>
|   |   member(s=todos, l=[dr,sr,de]) <line 23>
|   |   return false <line 9>
|   |   isUpper(s=los) <line 20>
|   |   return false <line 2>
|   |   member(s=los, l=[dr,sr,de]) <line 23>
|   |   return false <line 9>
|   |   isUpper(s=granujas) <line 20>
|   |   return false <line 2>
|   |   member(s=granujas, l=[dr,sr,de]) <line 23>
|   |   return false <line 9>
|   |   isUpper(s=del) <line 20>
|   |   return false <line 2>
|   |   member(s=del, l=[dr,sr,de]) <line 23>
|   |   return false <line 9>
|   |   isUpper(s=valle) <line 20>
|   |   return false <line 2>
|   |   member(s=valle, l=[dr,sr,de]) <line 23>
|   |   return false <line 9>
|   |   isUpper(s=,) <line 20>
|   |   return false <line 2>
|   |   member(s=,, l=[dr,sr,de]) <line 23>
|   |   return false <line 9>
|   |   isUpper(s=siempre) <line 20>
|   |   return false <line 2>
|   |   member(s=siempre, l=[dr,sr,de]) <line 23>
|   |   return false <line 9>
|   |   isUpper(s=ha) <line 20>
|   |   return false <line 2>
|   |   member(s=ha, l=[dr,sr,de]) <line 23>
|   |   return false <line 9>
|   |   isUpper(s=destacado) <line 20>
|   |   return false <line 2>
|   |   member(s=destacado, l=[dr,sr,de]) <line 23>
|   |   return false <line 9>
|   |   isUpper(s=el) <line 20>
|   |   return false <line 2>
|   |   member(s=el, l=[dr,sr,de]) <line 23>
|   |   return false <line 9>
|   |   isUpper(s=sr) <line 20>
|   |   return false <line 2>
|   |   member(s=sr, l=[dr,sr,de]) <line 23>
|   |   return true <line 6>
|   |   member(s=dr, l=[dr,sr,de]) <line 29>
|   |   return true <line 6>
|   |   member(s=Juan, l=[dr,sr,de]) <line 29>
|   |   return false <line 9>
|   |   isUpper(s=Juan) <line 32>
|   |   return true <line 2>
|   |   isUpper(s=de) <line 40>
|   |   return false <line 2>
|   |   member(s=de, l=[junior,senior]) <line 43>
|   |   return false <line 9>
|   |   member(s=de, l=[de,la,el]) <line 46>
|   |   return true <line 6>
|   |   member(s=la, l=[de,la,el]) <line 58>
|   |   return true <line 6>
|   |   member(s=Sierra, l=[de,la,el]) <line 58>
|   |   return false <line 9>
|   |   isUpper(s=Sierra) <line 61>
|   |   return true <line 2>
|   |   isUpper(s=junior) <line 40>
|   |   return false <line 2>
|   |   member(s=junior, l=[junior,senior]) <line 43>
|   |   return true <line 6>
|   |   member(s=como, l=[junior,senior]) <line 52>
|   |   return false <line 9>
|   return true <line 44>
return <line 70>
