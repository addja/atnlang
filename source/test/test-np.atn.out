Nombre encontrado: sr dr Juan de la Sierra junior 
